### PR TITLE
JSONRPCSerialiser 

### LINKED ISSUE #6: JSONRPCSerialiser
Class should provide interface to serialise and deserialise messages transported by [Relay Server API](https://docs.walletconnect.org/v/2.0/relay-server)

- messages should be deserialised into [Client Synchronisation Objects](https://docs.walletconnect.org/v/2.0/tech-spec#client-synchronization)
- [Client Synchronisation Objects](https://docs.walletconnect.org/v/2.0/tech-spec#client-synchronization) should be serialised into concatenated string iv+publicKey+mac+cipherText

### PR DESCRIPTION
**Pull request by llbartekll**
Closes #6

### CODE REVIEW

**Comment by llbartekll** [comment]
to add Serialiser tests with other Client Synchronisation types is in the scope of #8

**Comment by llbartekll** [comment]
to add Serialiser tests with new Client Synchronisation Types is in the scope of #8

File: Sources/WalletConnect_Swift/AES_256_CBC_HMAC_SHA256_Codec.swift [hunk 1]
```diff
@@ -30,95 +30,110 @@ struct HexString: Codable {
...
-        let keySha512 = keyData.sha512()
-        let key1 = keySha512.subdata(in: 0..<32)
-        let key2 = keySha512.subdata(in: 32..<64)
-        return (key1, key2)
-    }
+    static let ivLength = 16
+    static let publicKeyLength = 32
+    static let macLength = 32
 }
+//protocol Codec {
```
Comment (MisterVants) [inline Sources/WalletConnect_Swift/AES_256_CBC_HMAC_SHA256_Codec.swift:47]: Do we need to keep this commented code? Can we just remove it?

File: Sources/WalletConnect_Swift/Serialiser/JSONRPCSerialiserError.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// 
+
+import Foundation
+
+enum JSONRPCSerialiserError: String, Error, CustomStringConvertible {
+    case messageToShort = "Error: message is to short"
```
Comment (MisterVants) [inline Sources/WalletConnect_Swift/Serialiser/JSONRPCSerialiserError.swift:6]: There's a typo here in "message is 'too' short"

File: Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingApproveParams.swift [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// 
+
+import Foundation
+
+struct PairingApproveParams: Codable, Equatable {
+    let topic: String
+    
+    enum CodingKeys: CodingKey {
```
Comment (MisterVants) [inline Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingApproveParams.swift:8]: I believe we don't need to explicitly declare coding keys here.

File: Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingRejectParams.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// 
+
+import Foundation
+
+struct PairingRejectParams: Codable, Equatable {
+    let reason: String
+    
+    enum CodingKeys: CodingKey {
```
Comment (MisterVants) [inline Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingRejectParams.swift:8]: I believe we don't need to explicitly declare coding keys here.

File: Tests/WalletConnect_SwiftTests/AES_256_CBC_HMAC_SHA256_Codec_Test.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
+
+import Foundation
+import XCTest
+@testable import WalletConnect_Swift
+
+//class AES_256_CBC_HMAC_SHA256_Codec_Test: XCTestCase {
```
Comment (MisterVants) [inline Tests/WalletConnect_SwiftTests/AES_256_CBC_HMAC_SHA256_Codec_Test.swift:6]: Do we need to keep this commented test? Can we just remove it?

File: Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift [hunk 6]
```diff
@@ -26,57 +26,21 @@ struct JSONRPCError: Error, Codable {
     let message: String
 }
 
-struct JSONRPCRequest<T: Codable>: Encodable, JSONConvertible {
-    let id: Int64 = Self.generateId()
-    let jsonrpc = "2.0"
-    let method: String
+struct JSONRPCRequest<T: Codable, U: Codable>: Codable {
```
Comment (MisterVants) [inline Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift:29]: Do we still need this struct together with `ClientSynchJSONRPC`? They seem to be kind of duplicate, but maybe we can deal with this later.

File: Sources/WalletConnect_Swift/AES_256_CBC_HMAC_SHA256_Codec.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/WalletConnect_Swift/AES_256_CBC_HMAC_SHA256_Codec.swift:47]: It has been fixed in next PR :)

File: Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift (hunk 6 above)
Comment (llbartekll) [reply Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift:29]: Yes, we need this struct as `ClientSynchJSONRPC` is not the only JSON RPC request. I wanted to make it generic but had issues with decoding. But still I just realised that deserialising to `ClientSynchJSONRPC` won't be enough. Maybe you have some idea how to keep `JSONRPCRequest` generic and provide decoding inits for different types...

File: Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingApproveParams.swift (hunk 3 above)
Comment (llbartekll) [reply Sources/WalletConnect_Swift/Types/ClientSynchronisation/PairingApproveParams.swift:8]: you are absolutely right

File: Tests/WalletConnect_SwiftTests/AES_256_CBC_HMAC_SHA256_Codec_Test.swift (hunk 5 above)
Comment (llbartekll) [reply Tests/WalletConnect_SwiftTests/AES_256_CBC_HMAC_SHA256_Codec_Test.swift:6]: it is fixed in next PR

File: Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift (hunk 6 above)
Comment (MisterVants) [reply Sources/WalletConnect_Swift/JSONRPC/JSONRPC.swift:29]: I think I can handle this generic decoding later, let's go with what we need for now

**Review by MisterVants — APPROVED** [review]

**Review by MisterVants — APPROVED** [review]

**Pull request merged**